keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/Location.kt

```diff
@@ -20,6 +20,7 @@ package com.netflix.spinnaker.keel.api.ec2
 data class Location(
   val accountName: String,
   val region: String,
-  val subnet: String?, // TODO: is this actually optional?
+  val vpcName: String,
+  val subnet: String,
```

Comment:
VPC and subnet now required

Reply:
This seems wrong, since a `Subnet` always includes a `vpcId`.

Reply:
An alternative approach I considered was to look up the VPC based on the subnet when needed. Because we have subnets with no "purpose" property (which is what this `subnet` property maps to) I wasn't sure that would cover edge cases. Then again, I guess this doesn't either.

Reply:
I'm 95% sure we can ignore subnets without a purpose for now.

Reply:
how does this play with security groups not needing a subnet?

Reply:
They can just specify the VPC name or have it defaulted